src/hash/rescue/rpx/mod.rs

```diff
@@ -88,9 +91,16 @@ impl Hasher for Rpx256 {
         // every time the rate range is filled, a permutation is performed. if the final value of
         // `i` is not zero, then the chunks count wasn't enough to fill the state range, and an
         // additional permutation must be performed.
+        let mut current_element = 0_usize;
+        // handle the case of an empty `bytes`
+        let last_element = if num_field_elem == 0 {
+            current_element
+        } else {
+            num_field_elem - 1
+        };
         let i = bytes.chunks(BINARY_CHUNK_SIZE).fold(0, |i, chunk| {
```

Comment:
I would also probably rename `i` into something like `rate_pos` to make the naming explicit.

Reply:
Changed